pulp_rpm/app/advisory.py

```diff
@@ -25,6 +25,19 @@ def resolve_advisories(version, previous_version):
     """
     Decide which advisories to add to a repo version and which to remove, and adjust a repo version.
 
+    Advisory can be in 3 different states with relation to a repository version:
+     - in-memory and added before this function call, so it's a part of the current incomplete
+       repository version only
+     - in the db, it's been added in some previous repository version
+     - has no relation to any repository version because it's been created in this function as an
+       outcome of conflict resolution.
+
+    All 3 states need to be handled differently.
+    The in-db ones and newly created are straightforward, just remove/add in a standard way.
+    To remove in-memory ones (`content_pks_to_exclude`) from an incomplete repo version,
+    one needs to do it directly from RepositoryContent. They've never been a part of a repo
+    version, they are also not among the `content_pks_to_add` or `content_pks_to_remove` ones.
```

Comment:
It's probably not valid anymore, because this is now handled in the core. https://github.com/pulp/pulpcore/blame/master/pulpcore/app/models/repository.py#L602

We no longer need `content_pks_to_exclude` and `content_pks_to_remove`, just `content_pks_to_remove` should be  used for both cases. What do you think?

Reply:
+1 to any simplification we can do so long as we have decent tests to prevent regression

Reply:
ok, I'll file an issue thanks.